### DISCUSSION TITLE
ESP-NOW init done flag

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#110] [Course Qeustions] [answered]
Hi Mair,

With reagrd to the ESP-NOW routines, can you determine when ESP-NOW init is complete?
I want my code to wait until ESP-NOW is fully initialized before running other functions.

In the monitor window I notice the line below and thinking if I could somehow read this line in code for a flag that indicates it has been set up then this would do:

I [1712] ESPNOW: espnow [version 1.0] init

**Comment by gmag11** [comment, marked as answer]
I guess esp-now init is just a configuration process on esp32 internals, so it should be really fast. I've never had the need to wait after running init. Just read init result and expect ESP_OK if it was fine and ESP_ERR_ESPNOW_INTERNAL in case of error. I've never got that response though.

**Comment by Mair** [comment]
@orbitcoms There is no indication that initialization in non-blocking so when it returns. It should be initialized
